src/pkgmt/config.py

```diff
@@ -29,72 +30,155 @@ def __len__(self) -> int:
-        Function to return tag and push from
-        pyproject.toml if provided by user.
-        Example file:
+        Function to resolve tag and push values from CLI and config file.
+        CLI arguments take preference over values configured in config file.
+
+        Example pyproject.toml file:
         [tool.pkgmt]
         github = "repository/package"
         version = {version_file = "/app/_version.py", tag=True, push=False}
+
+        If values of tag or push is None from both CLI and config file, it is
+        set to default value True.
         """
-        tag = self.cfg.get("version", {}).get("tag", None)
-        push = self.cfg.get("version", {}).get("push", None)
-        if tag is not None and not isinstance(tag, bool):
-            raise InvalidConfiguration(
-                "Type of 'tag' key in pyproject.toml is invalid. "
-                "It should be lowercase boolean : true / false"
+
+        if cli_args is None:
+            cli_args = {}
+
+        cfg_version = data.get("version", {})
+        cfg_tag = cfg_version.get("tag", None)
+        cfg_push = cfg_version.get("push", None)
+
+        tag = cli_args.get("tag") if cli_args.get("tag") is not None else cfg_tag
+        push = cli_args.get("push") if cli_args.get("push") is not None else cfg_push
+
+        keys_to_validate = ["tag", "push"]
+        values_to_validate = [tag, push]
+
+        Config._validate_boolean_key(keys_to_validate, values_to_validate)
+
+        conflicts = []
+        if cfg_tag is not None and tag != cfg_tag:
+            conflicts.append(
+                Config._generate_overriding_error("tag", tag, cfg_tag, file)
             )
-        if push is not None and not isinstance(push, bool):
-            raise InvalidConfiguration(
-                "Type of 'push' key in pyproject.toml is invalid. "
-                "It should be lowercase boolean : true / false"
+        if cfg_push is not None and push != cfg_push:
+            conflicts.append(
+                Config._generate_overriding_error("push", push, cfg_push, file)
+            )
+
+        if conflicts:
+            click.echo("\n".join(conflicts))
+
+        tag = tag if tag is not None else True
+        push = push if push is not None else True
+
+        version = {"tag": tag, "push": push}
+        data.setdefault("version", {}).update(version)
+        return data
+
+    @classmethod
+    def from_file(cls, file, **kwargs):
+        """
+        Function to generate Config object from config file.
+        Config file should contain key tool.pkgmt
+        """
+        if Path(file).exists():
+            try:
+                with open(file) as f:
+                    data = toml.load(f)["tool"]["pkgmt"]
+                    Config._validate_config(data, file)
+                    data = Config._resolve_version_configuration(
+                        data, kwargs.get("cli_args"), file
+                    )
+                    return cls(data, file)
```

Comment:
when catching exceptions, it's better for the `try` statement to only contain the operation that you're looking to catch, otherwise other operations might raise errors that you don't want to catch.

in your case, you first want to try ... catch the `toml.load` operation and then you want to catch the `["tool"]["pkgmt"]` operation, you can split this so your `try` statements only contain whatever operation you want to check

Reply:
split the blocks. does this look correct?